keyboards/atreus/keymaps/ptillemans/keymap.c

```diff
@@ -0,0 +1,69 @@
+// this is the style you want to emulate.
+// This is the canonical layout file for the Quantum project. If you want to add another keyboard,
+
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _QW 0
+#define _RS 1
+#define _LW 2
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QW] = LAYOUT( /* Qwerty */
+    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                      KC_Y,    KC_U,    KC_I,    KC_O,    KC_P    ,
+    KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                      KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN ,
+    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                      KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH ,
+    TT(_LW), KC_TAB, KC_LGUI,  KC_LSFT, KC_BSPC,     F(2), KC_LALT, KC_SPC,  TT(_RS), KC_MINS, KC_QUOT, F(1)
+  ),
+
+  /*
+   *  !       @     up     {    }        ||     pgup    7     8     9    *
+   *  #     left   down  right  $        ||     pgdn    4     5     6    +
+   *  [       ]      (     )    &        ||       `     1     2     3    \
+   * lower  insert super shift bksp ctrl || alt space   fn    .     0    =
+   */
+  [_RS] = LAYOUT( /* [> RAISE <] */
+    KC_EXLM, KC_AT,   KC_UP,   KC_UNDS, KC_PLUS,                   KC_PGUP, KC_7,    KC_8,   KC_9, KC_ASTR ,
+    KC_HASH, KC_LEFT, KC_DOWN, KC_RGHT, KC_DLR,                    KC_PGDN, KC_4,    KC_5,   KC_6, KC_PLUS ,
+    KC_RBRC, KC_LBRC, KC_LPRN, KC_RPRN, KC_AMPR,                   KC_GRV,  KC_1,    KC_2,   KC_3, KC_BSLS ,
+    TT(_LW), _______, _______, _______, _______, _______, _______, _______, TO(_QW), KC_DOT,   KC_0, KC_EQL
+  ),
+  /*
+   * insert home   up  end   pgup       ||      up     F7    F8    F9   F10
+   *  del   left  down right pgdn       ||     down    F4    F5    F6   F11
+   *       volup             reset      ||             F1    F2    F3   F12
+   *       voldn  super shift next ctrl || alt space   L0  prtsc scroll pause
+   */
+  [_LW] = LAYOUT( /* [> LOWER <] */
+    KC_INS,  KC_HOME, KC_UP,   KC_END,  KC_PGUP,                   KC_UP,   KC_F7,   KC_F8,   KC_F9,   KC_F10  ,
+    KC_DELT, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN,                   KC_DOWN, KC_F4,   KC_F5,   KC_F6,   KC_F11  ,
+    KC_NO,   KC_VOLU, KC_NO,   KC_NO,   RESET,                     KC_TILD, KC_F1,   KC_F2,   KC_F3,   KC_F12  ,
+    KC_NO,   KC_VOLD, _______, _______, KC_MNXT, _______, _______, _______, TO(_QW), KC_PSCR, KC_SLCK, KC_MPLY
+  )
+};
+
+
+const uint16_t PROGMEM fn_actions[] = {
+  [1] = ACTION_MODS_TAP_KEY(MOD_LSFT, KC_ENT),
```

Comment:
QMK actually supports all these actions natively through a feature called [Mod Tap](https://docs.qmk.fm/#/feature_advanced_keycodes?id=mod-tap).

You can assign the following keycodes directly in your keymap above, and delete this code block (Lines 49-53; this function is also deprecated).

Instead of: | Use (in your keymap)
:------ | :------
`F(1)` | `MT(MOD_LSFT, KC_ENT)`
`F(2)` | `MT(MOD_LCTL, KC_ESC)`
`F(3)` | `MT(MOD_LALT, KC_SPC)`

Reply:
This. Emphatically.

We're actively trying to remove `fn_actions`, and `action_get_macro`,

Reply:
I agree, the new way is much better. I never updated it since I first created this keymap.